src/main/java/com/amalitech/gpuconfigurator/controller/CategoryController.java

```diff
@@ -0,0 +1,40 @@
+package com.amalitech.gpuconfigurator.controller;
+
+
+
+import com.amalitech.gpuconfigurator.dto.category.AllCategoryResponse;
+import com.amalitech.gpuconfigurator.dto.category.CategoryRequestDto;
+import com.amalitech.gpuconfigurator.model.Category;
+import com.amalitech.gpuconfigurator.service.CategoryService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api")
+public class CategoryController {
+
+    private final CategoryService categoryService;
+    @PostMapping("/v1/admin/category")
```

Comment:
By convention resource names should be plural in REST, so like `/admin/categories` not `/admin/category/`